src/providers/wms/qgswmsprovider.cpp

```diff
@@ -1068,6 +1068,11 @@ QUrl QgsWmsProvider::createRequestUrlWMS( const QgsRectangle &viewExtent, int pi
   setQueryItem( query, QStringLiteral( "HEIGHT" ), QString::number( pixelHeight ) );
   setQueryItem( query, QStringLiteral( "LAYERS" ), layers );
   setQueryItem( query, QStringLiteral( "STYLES" ), styles );
+  QStringList opacityList = mSettings.mOpacities;
+  if ( !opacityList.isEmpty() )
+  {
+    setQueryItem( query, QStringLiteral( "OPACITIES" ), mSettings.mOpacities.join( ',' ) );
```

Comment:
Random thought: is there a concept of escaping `,` characters in place?

Reply:
Good point. For opacities, it is not necessary, since it must be an int. For layers and styles it makes sense. I have to check the specs if WMS has something to escape commas in layer names.

Reply:
According to the spec, a comma in layername / stylename should be escaped with %2C. It seems neither the wms client in QGIS nor the server are prepared for that (the server would need to split the parameter value lists first and then unescape the commas in the single values).